### PR TITLE
enable category filter for courses

### PR DESCRIPTION
**Pull request by rawrrencee**
# Feature

Update Courses Category Menu to filter getCourses result by the category selected

# Changelog:

add utils function to process categories into a menu
add code on courses page to filter the courses by categoryId

## Checklist:

- [ ] Merged latest develop
- [ ] PR title makes sense

### CODE REVIEW

File: src/router.js [hunk 1]
```diff
@@ -28,6 +28,11 @@ const routes = [
     Component: lazy(() => import('pages/courses')),
     exact: true,
   },
+  {
+    path: '/courses/category/:categoryId',
+    Component: lazy(() => import('pages/courses')),
+    exact: true,
+  },
```
Comment (natalieloke) [inline src/router.js:40]: will there be a similar path in the router for mentorships with the categoryId

File: src/components/Public/PublicCategoriesBar/index.js [hunk 2]
```diff
@@ -1,130 +1,29 @@
...
   const [selectedKeys, setSelectedKeys] = useState(store.get('app.menu.selectedKeys') || [])
+  const categories = useSelector(state => state.categories)
+  if (pathname.includes('/mentorships')) menuData = mapCategoriesToMenu(categories, 'mentorships')
+  else if (pathname.includes('/courses')) menuData = mapCategoriesToMenu(categories, 'courses')
 
   useEffect(() => {
     applySelectedKeys()
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [pathname, menuData])
+  }, [pathname, categories])
```
Comment (natalieloke) [inline src/components/Public/PublicCategoriesBar/index.js]: if this bar also handles mentorships can consider changing name to just CategoryBar

File: src/router.js (hunk 1 above)
Comment (rawrrencee) [reply src/router.js:40]: yup! ok will include

File: src/components/Public/PublicCategoriesBar/index.js (hunk 2 above)
Comment (rawrrencee) [reply src/components/Public/PublicCategoriesBar/index.js]: righttt ok thanks! will change

**Review by natalieloke — APPROVED** [review]

**Pull request merged**